AnkiDroid/src/main/java/com/ichi2/libanki/Card.kt

```diff
@@ -468,8 +467,7 @@ open class Card : Cloneable {
          * Returns [flags] with the 3 first bits set as in [flag]
          */
         fun setFlagInInt(flags: Int, flag: Int): Int {
-            Assert.that(0 <= flag, "flag to set is negative")
-            Assert.that(flag <= 7, "flag to set is greater than 7.")
+            if (flag !in 0..7) throw IllegalArgumentException("flag outside of expected [0, 7] interval")
```

Comment:
Can we use `require { }` here, so Kotlin can apply code contracts?

It's also succinct

https://kotlinlang.org/api/core/kotlin-stdlib/kotlin/require.html

```suggestion
            require(flag in 0..7) { "flag outside of expected [0, 7] interval" }
```

Reply:
Done!